core/View/SecurityPolicy.php

```diff
@@ -0,0 +1,92 @@
+<?php
+/**
+ * Matomo - free/libre analytics platform
+ *
+ * @link https://matomo.org
+ * @license http://www.gnu.org/licenses/gpl-3.0.html GPL v3 or later
+ *
+ */
+namespace Piwik\View;
+
+use Piwik\Config;
+
+/**
+ * Content Security Policy HTTP Header management class
+ *
+ */
+class SecurityPolicy
+{
+    /**
+     * The policies that will generate the CSP header.
+     * These are keyed by the directive.
+     *
+     * @var array
+     */
+    private $policies = array();
+
+    private $cspEnabled = true;
+    private $reportOnly = false;
+
+    /**
+     * Constructor.
+     */
+    public function __construct() {
+        $this->policies['default-src'] = "'self' 'unsafe-inline' 'unsafe-eval'";
+
+        $generalConfig = Config::getInstance()->General;
+        $this->cspEnabled = $generalConfig['csp_enabled'];
+        $this->reportOnly = $generalConfig['csp_report_only'];
+    }
```

Comment:
Not necessarily needed, but we maybe could use dependency injection were easily possible:

```suggestion
    public function __construct(Config $config) {
        $this->policies['default-src'] = "'self' 'unsafe-inline' 'unsafe-eval'";

        $generalConfig = $config->General;
        $this->cspEnabled = $generalConfig['csp_enabled'];
        $this->reportOnly = $generalConfig['csp_report_only'];
    }
```